install.sh

```diff
@@ -1001,17 +1001,11 @@ case "${SHELL}" in
     ;;
 esac
 
-# `which` is a shell function defined above.
-# shellcheck disable=SC2230
-if [[ "$(which brew)" != "${HOMEBREW_PREFIX}/bin/brew" ]]
-then
-  cat <<EOS
-- Run these three commands in your terminal to add Homebrew to your ${tty_bold}PATH${tty_reset}:
-    echo '# Set PATH, MANPATH, etc., for Homebrew.' >> ${shell_profile}
-    echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"' >> ${shell_profile}
+cat <<EOS
+- Run these two commands in your terminal to add Homebrew to your ${tty_bold}PATH${tty_reset}:
+    printf '\\neval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"\\n' >> ${shell_profile}
```

Comment:
```suggestion
    echo '\\neval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"' >> ${shell_profile}
```

is this equivalent? If so it seems preferable, if not: leave as-is.

Reply:
`printf` is actually preferred for this. The `echo` variant requires a `-e` flag to process the escape sequences, but since this command will be executed by the user, the `echo` that gets run on the user machine may or may not understand `-e`, depending on the user's shell and `PATH`.

In contrast, all shell-builtin and standalone `printf` variants I've seen handle escape sequences in a predictable fashion.

Reply:
> The `echo` variant requires a `-e` flag to process the escape sequences

Why not `echo` twice to produce the `\n` linebreak? It does not require the escape substitution.

```suggestion
    echo; echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"' >> ${shell_profile}
```

Reply:
Thanks @XuehaiPan! I think you're right that `(echo; echo...)` is probably the best solution:
* it works fine with `bash` and `zsh` user shells, for which `$shell_profile` should also be correctly detected
* it generally fails loudly with `fish` and other non-Bourne shells, which IMO is better than (with `/bin/bash...`) quietly adding something that may never be used